### PR TITLE
Add and use new read_file() to handle reads in a stable manner

### PR DESCRIPTION
**Pull request by miff2000**
File content reads were being handled differently in four separate
places in the original script. These have been switched to use a
standard function `read_file()` which performs the same content
clean-up and conversion every time, and copes with foreign file
encodings consistently.

`test_chartbuilder.py` has been updated to use the codecs.open()
method also

This is as a fix for #65

### CODE REVIEW

File: pyhelm/chartbuilder.py [hunk 1]
```diff
@@ -236,6 +233,15 @@ def get_helm_chart(self):
         self._helm_chart = helm_chart
         return helm_chart
 
+    def read_file(self, path):
+        '''
+        Open the file provided in `path` and strip any non-UTF8 characters.
+        Return back the cleaned content
+        '''
+        content = codecs.open(path, encoding='utf-8', errors='ignore').read()
```
Comment (yanivoliver) [inline pyhelm/chartbuilder.py]: You need to close the file descriptor. You can wrap the ```read``` call with a ```with``` block doing the ```codes.open``` call

File: pyhelm/chartbuilder.py [hunk 2]
```diff
@@ -236,6 +233,15 @@ def get_helm_chart(self):
         self._helm_chart = helm_chart
         return helm_chart
 
+    def read_file(self, path):
```
Comment (yanivoliver) [inline pyhelm/chartbuilder.py]: This can (and should) be a static method

Comment (miff2000) [reply pyhelm/chartbuilder.py]: No worries. Using a decorator like this you mean?
```python
    @staticmethod
    def read_file(self, path):
```

File: pyhelm/chartbuilder.py (hunk 1 above)
Comment (miff2000) [reply pyhelm/chartbuilder.py]: Thanks, I missed that. Added in the latest commit

File: pyhelm/chartbuilder.py (hunk 2 above)
Comment (miff2000) [reply pyhelm/chartbuilder.py]: I've gone with this and added to the last commit. Let me know if that works for you

File: pyhelm/chartbuilder.py [hunk 3]
```diff
@@ -236,6 +233,18 @@ def get_helm_chart(self):
         self._helm_chart = helm_chart
         return helm_chart
 
+    @staticmethod
+    def read_file(self, path):
```
Comment (yanivoliver) [inline pyhelm/chartbuilder.py]: When using staticmethods, there's no 'self' passed. Symmetrically, calls to this method are done not by referencing 'self', but by the class itself.

File: pyhelm/chartbuilder.py [hunk 4]
```diff
@@ -236,6 +233,18 @@ def get_helm_chart(self):
...
+        '''
+        Open the file provided in `path` and strip any non-UTF8 characters.
+        Return back the cleaned content
+        '''
+        with codecs.open(path, encoding='utf-8', errors='ignore') as fd:
+            content = fd.read()
+            fd.close()
```
Comment (yanivoliver) [inline pyhelm/chartbuilder.py]: There's no need to call ```fd.close()``` as the with block takes care of that.

File: pyhelm/chartbuilder.py (hunk 3 above)
Comment (miff2000) [reply pyhelm/chartbuilder.py]: Riiiight. I'll sort that. Apologies for having to school me here. I'm fairly new to Python

File: pyhelm/chartbuilder.py (hunk 4 above)
Comment (miff2000) [reply pyhelm/chartbuilder.py]: Thanks.. Amended now

File: pyhelm/chartbuilder.py (hunk 3 above)
Comment (miff2000) [reply pyhelm/chartbuilder.py]: New commits pushed

**Comment by dano0b** [comment]
Would be great if this finds its way soon since it would help people using Ansible with the helm module. For now I have to use the command module as alternative.

File: pyhelm/chartbuilder.py (hunk 3 above)
Comment (yanivoliver) [reply pyhelm/chartbuilder.py]: No worries :)

**Comment by yanivoliver** [comment]
@miff2000 - Thanks!

**Pull request merged**